mlflow/pipelines/steps/train.py

```diff
@@ -586,6 +653,17 @@ def objective(hyperparameter_args):
                         "log_model_explainability": False,
                     },
                 )
+                autologged_params = mlflow.get_run(run_id=tuning_run.info.run_id).data.params
+                for param_name, param_value in estimator_args.items():
+                    if param_name in autologged_params:
+                        if not self._is_equal(param_value, autologged_params[param_name]):
+                            _logger.warning(
+                                f"Failed to log parameter {param_name} due to "
+                                f"conflict. old_value: {autologged_params[param_name]} "
+                                f"new_value: {param_value} type: {type(param_value)}"
```

Comment:
Can we be a bit more specific about the source of the conflict and what parameter we're attempting to log? I.e. it should be clear that we're trying to log a parameter from the hyperparameter tuning search space but we can't because the model trained at that point within the search space had a different value for the parameter. We should encourage the user to make sure that they're passing `estimator_params` properly to their estimator.